perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -1579,6 +1679,25 @@ public Member expireMember(PerunSession sess, Member member) throws WrongReferen
 		Status oldStatus = member.getStatus();
 		getMembersManagerImpl().setStatus(sess, member, Status.EXPIRED);
 		member.setStatus(Status.EXPIRED);
+		// if member went from invalid/disabled to expired, they might not exist in some parentVos - create them, but don't
+		// update MemberOrganizations, etc...
+		if (oldStatus == Status.INVALID || oldStatus == Status.DISABLED) {
+			try {
+				User user = getPerunBl().getUsersManagerBl().getUserByMember(sess, member);
+				List<Vo> parentVos = getPerunBl().getVosManagerBl().getParentVos(sess, member.getVoId());
+
+				for (Vo vo : parentVos) {
+					Member newMember = perunBl.getMembersManagerBl().createMember(sess, vo, user);
```

Comment:
I think the try-catch should go inside the forcycle, if the member exists in parentVo already, we still want to check the other vos.

Reply:
Big oversight on my part, thank you.